lib/tasks/generateResourcesJson.js

```diff
@@ -0,0 +1,49 @@
+"use strict";
+
+// const log = require("@ui5/logger").getLogger("builder:tasks:generateResourcesJson");
+const resourceListCreator = require("../processors/resourceListCreator");
+
+const DEFAULT_EXCLUDES = [
+	"!**/.DS_Store",
+	"!**/.eslintrc",
+	"!**/.eslintignore",
+	"!**/.gitignore"
+];
+
+/**
+ * Task for creating a library resources.json, describing all productive resources in the library.
+ *
+ * @public
+ * @alias module:@ui5/builder.tasks.generateResourcesJson
+ * @param {object} parameters Parameters
+ * @param {module:@ui5/fs.DuplexCollection} parameters.workspace DuplexCollection to read and write files
+ * @param {module:@ui5/fs.AbstractReader} parameters.dependencies Reader or Collection to read dependency files
+ * @param {object} parameters.options Options
+ * @param {string} parameters.options.projectName Project name
+ * @returns {Promise<undefined>} Promise resolving with <code>undefined</code> once data has been written
+ */
+module.exports = async function({workspace, dependencies, options}) {
+	let resources;
+	if (workspace.byGlobSource) { // API only available on duplex collections
+		resources = await workspace.byGlobSource(["/resources/**/*.*", ...DEFAULT_EXCLUDES]);
+		// HACK add resources from internal writer of workspace
+		const writtenResources = await workspace._writer.byGlob(["/resources/**/*.*", ...DEFAULT_EXCLUDES]);
+		writtenResources.forEach((res) => {
+			if ( resources.indexOf(res) < 0 ) {
+				resources.push(res);
+			}
+		});
+	} else {
+		resources = await workspace.byGlob(["/resources/**/*.*", ...DEFAULT_EXCLUDES]);
+	}
+	const dependencyResources = await dependencies.byGlob("/resources/**/*.{js,json,xml,html,properties,library}");
+
```

Comment:
I have no idea, why I added the resources from dependencies in my initial commit.

Thinking about it now, the resources.json is a library-local thing (or component-local thing) and should not include the dependencies. They have their own resources.json.

Wanted to verify the impact by building `sap.ui.layout` with dependencies. But building it failed in `sap.ui.core`, because `sap.ui.core` has modules outside its namespace -> orphans.

Reply:
done with d97bbe8

Reply:
Update: Analysis of XMLViews and fragments requires the resources from dependencies. That was one reason t odd them.